sys/vane/ford.hoon

```diff
@@ -1448,6 +1449,28 @@
               compiler-cache.state
       compiler-cache-size
     ==
+  ::  +reset: purge all builds and send %incomplete messages
+  ::
+  ::    The contents of Ford's build tables indirectly references the standard
+  ::    library and other core code. When a kernel reset occurs, these copies
+  ::    go out of date, making all the entries invalid. To handle this, we
+  ::    delete everything and send a message to everyone who is expecting a
+  ::    build result. These clients are expected to reiterate their original
+  ::    requests on hearing such %incomplete messages, allowing us to
+  ::    subsequently build our state anew.
+  ::
+  ++  reset  ^+  [moves state]
+    =<  finalize
+    =/  incompletes
+      %+  turn  ~(tap by ducts.state)
+      |=  [=^duct *]
+      `move`[duct %give %made now %incomplete ~]
+    ::
+    =.  moves  (weld incompletes moves)
+    :: For some reason, setting it to *_state instead of *ford-state
```

Comment:
That's because `_some-value` sets the default to `some-value`. This is how we can do
```
++  mul
  |=  [a=_1 b=_1]
  ...
```

Reply:
Huh. Good to know.